webcommon/web.clientproject.api/src/org/netbeans/modules/web/clientproject/api/json/JsonFile.java

```diff
@@ -70,13 +70,13 @@
     private static final ContainerFactory CONTAINER_FACTORY = new ContainerFactory() {
 
         @Override
-        public Map createObjectContainer() {
-            return new LinkedHashMap();
+        public Map<Object, Object> createObjectContainer() {
```

Comment:
This is the implementation of the `org.json.simple.parser.ContainerFactory` interface. If the simple json library introduces generics, this might blow. I had a similar problem a few weeks ago. Someone implemented an interface from JNA and decided to stick generics onto it (at that time JNA used a rawtype). This broke as recent JNA versions introduced generics at that point and were incompatible with the generic signature. At this point for example `<String,Object>` could be a valid alternative.

Reply:
Done.

Reply:
Sorry - misunderstanding - that was _one_ option how the API of `org.json.simple.parser.ContainerFactory` could be modified by the authors. `CONTAINER_FACTORY` implements this interface:

```java
package org.json.simple.parser;

import java.util.List;
import java.util.Map;

public interface ContainerFactory {
    public Map createObjectContainer();
    public List creatArrayContainer();
}
```

You can't predict how the interface will evolve, so don't speculate about it. The authors of simple json can add generics any time and when they do, that might clash with the signature this change introduces. I think this is a very valid place to keep rawtypes (maybe suppress them).